scipy/stats/_axis_nan_policy.py

```diff
@@ -566,16 +566,17 @@ def hypotest_fun_out(*samples, **kwds):
                     # Behave as though there are no NaNs (even if there are)
                     contains_nan = [False] * len(samples)
 
+                any_contains_nan = not is_jax(xp) and any(contains_nan)
```

Comment:
This could use an explanatory comment to make this easier to follow for someone who isn't aware of the context.